providers/neo4j.js

```diff
@@ -0,0 +1,215 @@
+	}
+
+	/**
+	 * Deletes a Document from a Collection that matches a user determined ID *
+	 * @param {string} table Name of the Collection
+	 * @param {string} id ID of the document
+	 * @returns {Promise<void>}
+	 */
+	delete(table, id) {
+		return this.db.run(`MATCH (n:${table} {id: {id} }) DELETE n`, { id });
+	}
+
+	/**
+	 * Replaces a Document with a new Document specified by the user *
+	 * @param {string} table Name of the Collection
+	 * @param {Object} id The Filter used to select the document to update
+	 * @param {Object} doc The Document that replaces the matching document
+	 * @returns {Promise<void>}
+	 */
+	async replace(table, id, doc) {
+		const { data } = await this.get(table, id);
+		const object = this.constructFlatObject(util.mergeObjects(data, this.parseUpdateInput(doc)));
+		return this.db.run(`MATCH (n:${table} {id : {id} }) SET ${object} RETURN n`, { id });
+	}
+
+	/**
+	 * Update or insert a new value to all entries.
+	 * @param {string} table The name of the table.
+	 * @param {string} path The object to remove or a path to update.
+	 * @param {*} newValue The new value for the key.
+    */
+	async updateValue(table, path, newValue) {
+		const keys = await this.getKeys(table);
+		if (util.isObject(path) && typeof newValue === 'undefined') {
+			Promise.all(keys.map(async node => await this.updatebyID(table, node, path)));
+		} else if (typeof path === 'string' && typeof newValue !== 'undefined') {
+			Promise.all(keys.map(async node => await this.updatebyID(table, node, util.makeObject(path, newValue))));
+		} else {
+			throw new TypeError(`Expected an object as first parameter or a string and a non-undefined value. Got: ${typeof key} and ${typeof value}`);
+		}
+	}
+	/**
+	 * Remove a value or object from all entries.
+	 * @param {string} table The name of the table.
+	 * @param {string} path The object to remove or a path to update.
+	 */
+	async removeValue(table, path) {
+		const keys = await this.getKeys(table);
+		if (util.isObject(path) && typeof newValue === 'undefined') {
+			Promise.all(keys.map(node => this.remove(table, node, path)));
+		} else if (typeof path === 'string' && typeof newValue !== 'undefined') {
+			Promise.all(keys.map(node => this.remove(table, node, util.makeObject(path, null))));
+		} else {
+			throw new TypeError(`Expected an object as first parameter or a string and a non-undefined value. Got: ${typeof key} and ${typeof value}`);
+		}
+	}
+	/**
+	 * Deletes a Document from a Collection that matches a user determined ID *
+	 * @param {string} table Name of the Collection
+	 * @param {string} id ID of the document
+	 * @param {string} path Object to be removed
+     * @private
+	 */
+	async remove(table, id, path) {
+		const [entry] = this.db.run(`MATCH (n:${table} {id: {id} }) RETURN n`, { id }).then(data => data.records.map(node => node._fields[0].properties));
+		delete entry[[Object.keys(path)]];
```

Comment:
This doesn't seem valid to me... and this method's description describes what `delete()` does